### ISSUE TITLE
Current Playing Notifications

### ISSUE THREAD

**Issue by SethGower** [Rigellute/spotify-tui#146]
A feature I really enjoy in the official Spotify Client, which I would like to see in `spt` is the current playing notification. It includes the song title, artist and album art. I have been debating on which service to implement this in, whether it should go in `spt` or `spotifyd`, and figure the best and most efficient way of implementing it would be in `spt` since all of the track information is already available, and there is already an API key entangled with the service, allowing for the necessary API calls. If it were implemented in `spotifyd` instead, then a new API key would need to be generated in order to get the current playing art, info, etc. I am making this issue to solely to have this recorded somewhere, but I plan on implementing this in the nearish future. Also wanted to get any input from other contributors if you all had any other ideas/suggestions. Here is an example of what I am talking about (pardon the Christmas music in November). 
![image](https://user-images.githubusercontent.com/19225089/68553289-d42be380-03ed-11ea-9573-51d271255d8c.png)

**Comment by Rigellute** [comment]
Interesting idea - are these operating system notifications?

**Comment by SethGower** [comment]
@Rigellute Yes. I was planning on implementing using `libnotify`, as Spotify client does

**Comment by SethGower** [comment]
I have finished creating the function to actually send the notification. It looks Almost identical to the notification that spotify serves. 
![image](https://user-images.githubusercontent.com/19225089/68677905-e495a880-052a-11ea-83a8-a899b79784e2.png)

Now that I have it working, I could use some help on figuring out where to actually implement it. If you want to see the code, it is [here](https://github.com/SethGower/spotify-tui/commit/233715aeebd60ad3f98311a94e6929d19f54d8e9). I need to find a way to only call the function when the song has changed/when the state has changed. Any suggestions?

**Comment by Rigellute** [comment]
There's currently no elegant way of checking if the song has changed.

Every 5 seconds we poll the spotify API for current playback context. After certain events (like skipping a track) we also attempt to fetch playback context, but there is a race here so the response is not always accurate.

I would put the "has song changed" logic in `get_current_playback` function here https://github.com/Rigellute/spotify-tui/blob/master/src/app.rs#L332

Check if the local playback context is different from the context we've fetched from the API (use the track id).

This would be something like `self.current_playback_context.item.track.id` (although you'd need to handle the Options in this chain). 

Hope that makes sense?

**Comment by SethGower** [comment]
Ok, that was what I was thinking. I figured I would just mention it, in case I missed it, given I am still becoming familiar with rust. Thanks for all of the help!
